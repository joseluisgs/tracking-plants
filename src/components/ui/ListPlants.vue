<template>
  <div>
    <h1 class="title font-bold text-xl mb-4">Mis Plantas</h1>
    <ul>
      <li>
        <button
          class="border border-black rounded-full py-1 px-3 text-3xl"
          @click="goToAddPlant"
        >
          +
        </button>
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  name: 'ListPlants',
  methods: {
    goToAddPlant() {
      this.$router.push({ name: 'AddPlant' });
    },
  },
};
</script>
<style>
</style>
