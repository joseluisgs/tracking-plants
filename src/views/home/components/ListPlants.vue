<template>
  <div>
     <h1 class="title font-bold text-xl mb-4">
      Mis plantas:
    </h1>
    <ul class="list flex items-center overflow-x-scroll py-4">
      <li class="mr-5">
         <!-- data-test-id Esta clase solo nos sirve para detectar el botÃ³n en el test -->
        <button
          data-test-id="add-button"
          class="add-button"
          @click="goToAddPlant"
        >
          +
        </button>
      </li>
      <!-- Listado de plantas -->
      <li
        v-for="plant in plants"
        :key="plant.id"
        class="plant-item hover:bg-gray-100;"
        @click="clickItem(plant.id)"
        data-test="list-plants-item"
      >
      <!-- src="@/assets/images/botanical.svg" -->
        <img
          class="w-24 h-24 rounded-lg"
          :src="plant.image"
          :alt="plant.name"
        />
        <p class="text-sm capitalize font-semibold">{{ plant.name }}</p>
        <p class="duration"> {{plant.dueDate | moment("D MMMM YYYY")}}</p>
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  name: 'ListPlants',
  // Mis propiedades
  props: {
    plants: {
      type: Array,
      default: () => [],
    },
  },
  methods: {
    goToAddPlant() {
      // navegamos a...
      this.$router.push({ name: 'AddPlant' });
    },
    clickItem(plantId) {
      this.$emit('item-clicked', plantId);
    },
  },
};
</script>

<style lang="scss" scoped>
  // Para evitar que salga el scroll
  .list {
    -ms-overflow-style: none;
    scrollbar-width: none;
    &::-webkit-scrollbar {
      display: none;
    }
  }
  .plant-item {
    min-width: 140px;
    max-width: 140px;
    min-height: 140px;
    max-height: 140px;
    @apply shadow-lg;
    @apply w-32;
    @apply h-32;
    @apply rounded-lg;
    @apply mr-5;
    @apply flex;
    @apply items-center;
    @apply justify-center;
    @apply flex-col;
    @apply cursor-pointer;
    @apply bg-green-200;
  }
  .duration {
    width: fit-content;
    @apply px-4;
    @apply py-1;
    @apply font-medium;
    @apply text-xs;
    @apply text-center;
  }
  .add-button {
    border-radius: 16px;
    line-height: unset;
    @apply flex;
    @apply justify-center;
    @apply border;
    @apply text-3xl;
    @apply w-12;
    @apply h-12;
    @apply bg-green-500
}
</style>
