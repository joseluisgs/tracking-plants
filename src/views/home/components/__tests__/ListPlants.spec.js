import { shallowMount } from '@vue/test-utils';
import ListPlants from '../ListPlants.vue';

// Teteamos su componente
describe('Vista: HomeView -> Componente: ListPlants', () => {
  const $router = {
    push: jest.fn(),
  };

  // Crea el shallowMount
  const build = () => {
    const wrapper = shallowMount(ListPlants, {
      mocks: {
        $router,
      },
    });

    return {
      wrapper,
    };
  };

  test('Debería ser igual a su Snapshot', () => {
    const { wrapper } = build();
    expect(wrapper.element).toMatchSnapshot();
  });

  it('Debería ir a la ruta indicada al hacer click', () => {
    const { wrapper } = build();
    // Buscamos el botón etiquetado de esta manera
    const button = wrapper.find('[data-test-id="add-button"]');
    // Lo disparamos
    button.trigger('click');
    // Se ha llamado una sola vez
    expect(wrapper.vm.$router.push).toHaveBeenCalledTimes(1);
    // Y va a esa ruta
    expect(wrapper.vm.$router.push).toHaveBeenCalledWith({ name: 'AddPlant' });
  });
});
