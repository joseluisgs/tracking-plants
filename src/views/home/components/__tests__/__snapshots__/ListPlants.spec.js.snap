// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vista: HomeView -> Componente: ListPlants Deber√≠a ser igual a su Snapshot 1`] = `
<div>
  <h1
    class="title font-bold text-xl mb-4"
  >
    
    Mis plantas:
  
  </h1>
   
  <ul
    class="list flex items-center overflow-x-scroll py-4"
  >
    <li
      class="mr-5"
    >
      <button
        class="flex justify-center rounded-lg border border-red-main-400 text-3xl w-12 h-12 bg-green-600 text-white"
        data-test-id="add-button"
      >
        
        +
      
      </button>
    </li>
     
  </ul>
</div>
`;
